### DISCUSSION TITLE
Best practices for using logstash-logback-encoder?

### DISCUSSION THREAD

**Discussion by wsargent** [logfellow/logstash-logback-encoder#481] [Show and tell]
NOTABUG, but it doesn't look like you have a development mailing list.

I wrote up a project on doing [structured logging using logstash-logback-encoder](https://github.com/tersesystems/terse-logback), and have been wondering if there are any other examples or written guidelines out there.  Is there a repository for this?

**Comment by philsttr** [comment]
Cool article.  I'm not aware of any guidelines, but there are a bunch of blog posts on basic usage.

**Comment by wsargent** [comment]
@philsttr Thanks!  I've been picking up more about logback as I go along -- currently I'm figuring out how to get a composite converter be able to reference other entities in context through the options list. 

Are there any logback extension projects you'd recommend as good places to learn more about logback?

**Comment by wsargent** [comment]
Wrote some blog posts:

https://tersesystems.com/blog/2019/04/23/application-logging-in-java-part-1/

**Comment by wsargent** [comment]
Benchmarks: https://github.com/wsargent/slf4j-benchmark/blob/master/README.md

**Comment by wsargent** [comment]
SHowcase: https://terse-logback-showcase.herokuapp.com/

**Comment by wsargent** [comment]
Scala logging API with logstash support: https://tersesystems.github.io/blindsight/usage/overview.html

**Comment by wsargent** [comment]
https://github.com/tersesystems/blindsight is 1.0.0
